### DISCUSSION TITLE
CC1101 module not found

### DISCUSSION THREAD

**Discussion by cterwilliger** [jgromes/RadioLib#428] [Q&A] [answered]
I'm trying out a Solu CC1101 module (Amazon) with a Heltec WiFi Kit 32 (ESP32).  Using the library example for CC1101 receive with nothing changed but the pins.  I traced the pins from the header to the CC1101 chip so I know they are correct.  Using GPIO2 for GDO0 and GPIO18 for CSN.  

[CC1101] Initializing ... CC1101 not found! (1 of 10 tries) RADIOLIB_CC1101_REG_VERSION == 0x0000, expected 0x0004/0x0014
CC1101 not found! (2 of 10 tries) RADIOLIB_CC1101_REG_VERSION == 0x0000, expected 0x0004/0x0014
CC1101 not found! (3 of 10 tries) RADIOLIB_CC1101_REG_VERSION == 0x0000, expected 0x0004/0x0014
CC1101 not found! (4 of 10 tries) RADIOLIB_CC1101_REG_VERSION == 0x0000, expected 0x0004/0x0014
CC1101 not found! (5 of 10 tries) RADIOLIB_CC1101_REG_VERSION == 0x0000, expected 0x0004/0x0014
CC1101 not found! (6 of 10 tries) RADIOLIB_CC1101_REG_VERSION == 0x0000, expected 0x0004/0x0014
CC1101 not found! (7 of 10 tries) RADIOLIB_CC1101_REG_VERSION == 0x0000, expected 0x0004/0x0014
CC1101 not found! (8 of 10 tries) RADIOLIB_CC1101_REG_VERSION == 0x0000, expected 0x0004/0x0014
CC1101 not found! (9 of 10 tries) RADIOLIB_CC1101_REG_VERSION == 0x0000, expected 0x0004/0x0014
CC1101 not found! (10 of 10 tries) RADIOLIB_CC1101_REG_VERSION == 0x0000, expected 0x0004/0x0014
No CC1101 found!
failed, code -2

Any thoughts?  Hopefully my $7 cheap module isn't garbage...

**Comment by jgromes** [comment, marked as answer]
The module is returning 0x00 for any SPI read, so the wiring is where I would start - power and SPI in particular. Next, there's the SPI configruation. I don't remember which SPI bus (HSPI or VSPI) is used by default on ESP32, so it's better to specifically define your own `SPIClass` instance.

**Reply by cterwilliger** [reply]
Had a wiring error. For anyone else using these, note that the Heltec modules use HSPI for the OLED display, so you only have VSPI available.
